godot-core/src/builtin/dictionary.rs

```diff
@@ -441,6 +445,11 @@ impl<'a> DictionaryIter<'a> {
         Some((key, value))
     }
 
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        let remaining = self.dictionary.len() - self.next_idx;
+        (remaining, Some(remaining))
+    }
```

Comment:
What happens when an element is _removed_ while iterating, and you are at the last element?
Would make an interesting test case, too 😉 

Basically this subtraction could underflow.

Reply:
Subtracting now with `overflowing_sub()` to cover this case :)